teuthology/suite/__init__.py

```diff
@@ -200,7 +208,106 @@ def get_rerun_conf_overrides(conf):
         )
         return
 
-    conf.filter_in.extend(rerun_filters['descriptions'])
+    if getattr(conf, 'skip_known_failures', False):
+        # Filter jobs by rerun statuses first
+        rerun_statuses_jobs = (job for job in run['jobs'] if job['status'] in conf.rerun_statuses)
+        # Filter out known failures
+        known_patterns_file = getattr(conf, 'known_patterns_file', None)
+        rerun_jobs = filter_out_known_failures(rerun_statuses_jobs, known_patterns_file)
+        if rerun_jobs:
+            conf.filter_in.extend(job['description'] for job in rerun_jobs if job.get('description'))
+            log.info("Using unknown failure filtering for rerun with %d unknown failures", len(rerun_jobs))
+        else:
+            log.info("No unknown failures found, skipping rerun")
+            return False
+    else:
+        original_descriptions = [job['description'] for job in run['jobs'] if job['status'] in conf.rerun_statuses and job['description']]
+        conf.filter_in.extend(original_descriptions)
+        log.info("Using backward-compatible job filtering for rerun with %d job descriptions", len(original_descriptions))
+
+
+def _get_default_known_patterns_file():
+    """Returns the path to the default known patterns file bundled with teuthology.
+    
+    The file format is documented in docs/siteconfig.rst.
+    """
+    return os.path.join(os.path.dirname(__file__), 'patterns', 'known-failures.json')
+
+
+def _load_known_patterns(known_patterns_file):
+    """Load known failure patterns from a JSON or YAML file.
+    
+    Args:
+        known_patterns_file: Path to the patterns file. If None, uses the default
+                           bundled file.
+    
+    Returns:
+        List of regex patterns (strings) to match against failure reasons.
+    
+    The file format should be:
+        JSON: {"patterns": ["pattern1", "pattern2", ...]}
+        YAML: patterns: ["pattern1", "pattern2", ...]
```

Comment:
Thanks for adding docstring for the function, but could you please stick to the conventions being used with teuthology, like use `:param param_name:   param description` and `:returns: descriptions what returns` etc. For example, refer other functions in teuthology code base.